### PR TITLE
VMSS Flex Support: part 5 of FlexScaleSet Implementation of VMSet interface mainly for network related functions

### PR DESCRIPTION
**Pull request by zmyzheng**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This PR is the part 5 of FlexScaleSet implementation of VMSet, mainly for network related functions
#### Which issue(s) this PR fixes:

Refer to #639 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

**Comment by nilo19** [comment]
Is it possible to enrich the unit test for this one?

**Comment by zmyzheng** [comment]
> Is it possible to enrich the unit test for this one?

Unit tests are added for all the newly added functions now.

File: pkg/provider/azure_vmssflex_cache.go [hunk 1]
```diff
@@ -123,6 +123,45 @@ func (fs *FlexScaleSet) newVmssFlexVMCache() (*azcache.TimedCache, error) {
...
+	getter := func(vmName string, crt azcache.AzureCacheReadType) (string, error) {
+		cached, err := fs.vmssFlexCache.Get(consts.VmssFlexKey, crt)
+		if err != nil {
+			return "", err
+		}
+		vmssFlexes := cached.(*sync.Map)
+
+		vmssFlexes.Range(func(key, value interface{}) bool {
+			vmssFlexID := key.(string)
+			_, err := fs.vmssFlexVMCache.Get(vmssFlexID, azcache.CacheReadTypeForceRefresh)
```
Comment (andyzhangx) [inline pkg/provider/azure_vmssflex_cache.go:143]: this is also flushing cache in every `getNodeNameByVMName` call, why not add `crt azcache.AzureCacheReadType` as a parameter, and then set `azcache.CacheReadTypeDefault` as default in the caller?

Comment (zmyzheng) [reply pkg/provider/azure_vmssflex_cache.go:143]: If the code goes to the line you marked, that means isCached is false in line 130, so we need to forcely refresh the cache.

File: pkg/provider/azure_vmssflex_cache.go [hunk 2]
```diff
@@ -123,6 +123,45 @@ func (fs *FlexScaleSet) newVmssFlexVMCache() (*azcache.TimedCache, error) {
...
+			if err != nil {
+				klog.V(12).Infof("failed to refresh vmss flex VM cache for vmssFlexID %s", vmssFlexID)
```
Comment (andyzhangx) [inline pkg/provider/azure_vmssflex_cache.go:145]: error logs

File: pkg/provider/azure_vmssflex_cache.go [hunk 3]
```diff
@@ -123,6 +123,45 @@ func (fs *FlexScaleSet) newVmssFlexVMCache() (*azcache.TimedCache, error) {
...
+		cachedNodeName, isCached = fs.vmssFlexVMNameToNodeName.Load(vmName)
+		if isCached {
+			return fmt.Sprintf("%v", cachedNodeName), nil
+		}
+		return "", cloudprovider.InstanceNotFound
+	}
+
+	nodeName, err := getter(vmName, azcache.CacheReadTypeDefault)
+	if err == cloudprovider.InstanceNotFound {
+		return getter(vmName, azcache.CacheReadTypeForceRefresh)
```
Comment (andyzhangx) [inline pkg/provider/azure_vmssflex_cache.go:159]: add a `klog.V(2).Infof` logging before cache refresh, that would benefit in troubleshooting

File: pkg/provider/azure_vmssflex_cache.go (hunk 2 above)
Comment (andyzhangx) [reply pkg/provider/azure_vmssflex_cache.go:145]: btw, when this func would return false?

**Comment by nilo19** [comment]
> > Is it possible to enrich the unit test for this one?
> 
> Unit tests are added for all the newly added functions now.

Can we cover more new lines in this one as well?

File: pkg/provider/azure_vmssflex_cache.go (hunk 2 above)
Comment (zmyzheng) [reply pkg/provider/azure_vmssflex_cache.go]: This is removed after rebase due to code duplication

Comment (zmyzheng) [reply pkg/provider/azure_vmssflex_cache.go]: This func would not return false.

File: pkg/provider/azure_vmssflex_cache.go (hunk 3 above)
Comment (zmyzheng) [reply pkg/provider/azure_vmssflex_cache.go]: This is removed after rebase due to code duplication

**Comment by zmyzheng** [comment]
> > > Is it possible to enrich the unit test for this one?
> > 
> > 
> > Unit tests are added for all the newly added functions now.
> 
> Can we cover more new lines in this one as well?

More Unit tests are added to cover new lines

**Pull request merged**